plugins/inventory/aws_ec2.py

```diff
@@ -757,46 +776,44 @@ def _add_hosts(self, hosts, group, hostnames, allow_duplicated_hosts=False):
             else:
                 continue
 
-            host = camel_dict_to_snake_dict(host, ignore_list=['Tags'])
-            host['tags'] = boto3_tag_list_to_ansible_dict(host.get('tags', []))
+            host_vars = self.prepare_host_vars(
+                host,
+                hostvars_prefix,
+                hostvars_suffix,
+                use_contrib_script_compatible_ec2_tag_keys)
+            for name in hostname_list:
+                yield to_text(name), host_vars
 
-            if self.get_option('use_contrib_script_compatible_ec2_tag_keys'):
-                for k, v in host['tags'].items():
-                    host["ec2_tag_%s" % k] = v
-
-            # Allow easier grouping by region
-            host['placement']['region'] = host['placement']['availability_zone'][:-1]
+    def _add_hosts(self, hosts, group, hostnames, allow_duplicated_hosts=False, hostvars_prefix=None, hostvars_suffix=None, use_contrib_script_compatible_ec2_tag_keys=False):
+        '''
+            :param hosts: a list of hosts to be added to a group
+            :param group: the name of the group to which the hosts belong
+            :param hostnames: a list of hostname destination variables in order of preference
+        '''
 
-            if not hostname_list:
-                continue
-            for hostname in hostname_list:
-                self.inventory.add_host(to_text(hostname), group=group)
-            hostvars_prefix = self.get_option("hostvars_prefix")
-            hostvars_suffix = self.get_option("hostvars_suffix")
-            new_vars = dict()
-            for hostvar, hostval in host.items():
-                if hostvars_prefix:
-                    hostvar = hostvars_prefix + hostvar
-                if hostvars_suffix:
-                    hostvar = hostvar + hostvars_suffix
-                new_vars[hostvar] = hostval
-                for hostname in hostname_list:
-                    self.inventory.set_variable(to_text(hostname), hostvar, hostval)
-            host.update(new_vars)
+        for name, host_vars in self.iter_entry(
+                hosts, hostnames,
+                allow_duplicated_hosts=allow_duplicated_hosts,
+                hostvars_prefix=hostvars_prefix,
+                hostvars_suffix=hostvars_suffix,
+                use_contrib_script_compatible_ec2_tag_keys=use_contrib_script_compatible_ec2_tag_keys):
+            self.inventory.add_host(name, group=group)
+            assert isinstance(host_vars, dict)
```

Comment:
https://docs.ansible.com/ansible/latest/dev_guide/testing/sanity/no-assert.html

Reply:
oh, it was supposed to go away before I push the PR for review. Let's me fix that.